### PR TITLE
[Fix] SupportsFilterPushDown bug

### LINKED ISSUE #478: [Bug] SupportsFilterPushDown produces incorrect results when merging the result sets of different filter conditions on the same table
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

all

### What's Wrong?

After running different filter conditions separately on the same table and merging the two results with UNION ALL, a concatenation error occurred when applying the filterdown condition.

Exmaple:
image

Generated filtered sql

image

### What You Expected?

operating normally

### How to Reproduce?

_No response_

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by MaoMiMao**
# Proposed changes

Issue Number: close #478 

## Problem Summary:

image

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by JNSimba** [comment]
Please add a union all case

File: flink-doris-connector/src/main/java/org/apache/doris/flink/source/DorisSource.java [hunk 1]
```diff
@@ -95,6 +100,15 @@ public SourceReader<OUT, DorisSourceSplit> createReader(SourceReaderContext read
     public SplitEnumerator<DorisSourceSplit, PendingSplitsCheckpoint> createEnumerator(
             SplitEnumeratorContext<DorisSourceSplit> context) throws Exception {
         List<DorisSourceSplit> dorisSourceSplits = new ArrayList<>();
+        if (!resolvedFilterQuery.isEmpty()) {
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/source/DorisSource.java:103]: Is it possible that there is a NPE here?

File: flink-doris-connector/src/test/java/org/apache/doris/flink/source/DorisSourceITCase.java [hunk 2]
```diff
@@ -67,6 +67,7 @@ public void testSource() throws Exception {
         DorisSource<List<?>> source =
                 DorisSource.<List<?>>builder()
                         .setDorisOptions(dorisBuilder.build())
+                        .setResolvedFilterQuery(new ArrayList<>())
```
Comment (JNSimba) [inline flink-doris-connector/src/test/java/org/apache/doris/flink/source/DorisSourceITCase.java:70]: Maybe you can just give a default value

File: flink-doris-connector/src/main/java/org/apache/doris/flink/source/DorisSource.java [hunk 3]
```diff
@@ -95,6 +100,15 @@ public SourceReader<OUT, DorisSourceSplit> createReader(SourceReaderContext read
...
+            String collect = String.join(" AND ", resolvedFilterQuery);
```
Comment (JNSimba) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/source/DorisSource.java:104]:
```suggestion
            String filterQuery = String.join(" AND ", resolvedFilterQuery);
```

File: flink-doris-connector/src/test/java/org/apache/doris/flink/source/DorisSourceITCase.java (hunk 2 above)
Comment (MaoMiMao) [reply flink-doris-connector/src/test/java/org/apache/doris/flink/source/DorisSourceITCase.java:70]: GET

File: flink-doris-connector/src/main/java/org/apache/doris/flink/source/DorisSource.java (hunk 1 above)
Comment (MaoMiMao) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/source/DorisSource.java:103]: I will set an initial value

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**